system-addon/lib/SearchFeed.jsm

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+ /* globals ContentSearch, XPCOMUtils, Services */
+"use strict";
+
+const {utils: Cu} = Components;
+const {actionTypes: at, actionCreators: ac} = Cu.import("resource://activity-stream/common/Actions.jsm", {});
+const SEARCH_ENGINE_TOPIC = "browser-search-engine-modified";
+const CONTENT_SEARCH_READY = "content-search-initialized";
+
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "ContentSearch",
+  "resource:///modules/ContentSearch.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "Services",
+  "resource://gre/modules/Services.jsm");
+
+this.SearchFeed = class SearchFeed {
+  addObservers() {
+    Services.obs.addObserver(this, SEARCH_ENGINE_TOPIC, false);
+    Services.obs.addObserver(this, CONTENT_SEARCH_READY, false);
+  }
+  removeObservers() {
+    Services.obs.removeObserver(this, SEARCH_ENGINE_TOPIC);
+    Services.obs.removeObserver(this, CONTENT_SEARCH_READY);
+  }
+  observe(subject, topic, data) {
+    switch (topic) {
+      case CONTENT_SEARCH_READY:
+        this.getState();
```

Comment:
@k88hudson The issue we're running into is the add-on when installed and enabled as a system add-on runs its init related code before the rest of the browser -- in this specific instance depending on `ContentSearch.init()`, which runs from `finalUIStartup` of `nsBrowserGlue`, which also includes other new tab module inits:

https://dxr.mozilla.org/mozilla-central/source/browser/components/nsBrowserGlue.js#507-512

One potential solution is to have the add-on only trigger `bootstrap.js`'s `ActivityStream.init` after it has observed `browser-ui-startup-complete`, which is notified at the end of `finalUIStartup`. However by delaying initialization, there's a risk of being too late?

I suppose the usual way if activity stream was just another module in Firefox, it would get its init triggered directly from `finalUIStartup`.